ledger/light/proc/executor_test.go

```diff
@@ -0,0 +1,132 @@
+	"github.com/stretchr/testify/assert"
+
+	"github.com/insolar/insolar/insolar"
+	"github.com/insolar/insolar/insolar/bus"
+	"github.com/insolar/insolar/insolar/gen"
+	"github.com/insolar/insolar/insolar/jet"
+	"github.com/insolar/insolar/insolar/payload"
+	"github.com/insolar/insolar/instrumentation/inslogger"
+	"github.com/insolar/insolar/ledger/light/executor"
+	"github.com/insolar/insolar/ledger/light/proc"
+)
+
+func TestFetchJet_Proceed(t *testing.T) {
+	ctx := inslogger.TestContext(t)
+	mc := minimock.NewController(t)
+
+	var (
+		jetAccessor *jet.AccessorMock
+		jetFetcher  *executor.JetFetcherMock
+		coordinator *jet.CoordinatorMock
+		sender      *bus.SenderMock
+	)
+	setup := func() {
+		jetAccessor = jet.NewAccessorMock(mc)
+		jetFetcher = executor.NewJetFetcherMock(mc)
+		coordinator = jet.NewCoordinatorMock(mc)
+		sender = bus.NewSenderMock(mc)
+
+	}
+
+	t.Run("error wrong jetID", func(t *testing.T) {
+		setup()
+		defer mc.Finish()
+
+		jetFetcher.FetchMock.Return(&insolar.ID{}, errors.New("test"))
+
+		p := proc.NewFetchJet(gen.ID(), insolar.FirstPulseNumber, payload.Meta{}, false)
+		p.Dep(jetAccessor, jetFetcher, coordinator, sender)
+		err := p.Proceed(ctx)
+		assert.Error(t, err, "expected error 'failed to fetch jet'")
+	})
+
+	t.Run("virtual is executor", func(t *testing.T) {
+		setup()
+		defer mc.Finish()
+
+		id := gen.ID()
+		jetFetcher.FetchMock.Return(&id, nil)
+		me := insolar.NewReference(gen.ID())
+		coordinator.LightExecutorForJetMock.Return(me, nil)
+		coordinator.MeMock.Return(*me)
+
+		p := proc.NewFetchJet(gen.ID(), insolar.FirstPulseNumber, payload.Meta{}, false)
+		p.Dep(jetAccessor, jetFetcher, coordinator, sender)
+		err := p.Proceed(ctx)
+		assert.NoError(t, err)
+		assert.Equal(t, p.Result.Jet, insolar.JetID(id))
+	})
+
+	t.Run("virtual is not executor", func(t *testing.T) {
+		setup()
+		defer mc.Finish()
+
+		id := gen.ID()
+		jetFetcher.FetchMock.Return(&id, nil)
+		me := insolar.NewReference(gen.ID())
+		notMe := insolar.NewReference(gen.ID())
+		coordinator.LightExecutorForJetMock.Return(me, nil)
+		coordinator.MeMock.Return(*notMe)
+
+		p := proc.NewFetchJet(gen.ID(), insolar.FirstPulseNumber, payload.Meta{}, false)
+		p.Dep(jetAccessor, jetFetcher, coordinator, sender)
+		err := p.Proceed(ctx)
+		assert.Error(t, err)
+	})
+
+	t.Run("virtual passing", func(t *testing.T) {
+		setup()
+		defer mc.Finish()
+
+		id := gen.ID()
+		jetFetcher.FetchMock.Return(&id, nil)
+		msgSender := insolar.NewReference(gen.ID())
+		me := insolar.NewReference(gen.ID())
+		notMe := insolar.NewReference(gen.ID())
+		coordinator.LightExecutorForJetMock.Return(notMe, nil)
+		coordinator.MeMock.Return(*me)
+
+		reps := make(chan *message.Message, 1)
+
+		sender.SendTargetMock.Inspect(func(ctx context.Context, msg *message.Message, target insolar.Reference) {
+			res := false
+			if target == *notMe || target == *msgSender {
+				res = true
+			}
+			assert.True(t, res, "expected messages to msgSender and to right executor")
```

Comment:
`assert.True(t, target == *notMe || target == *msgSender, "expected messages to msgSender and to right executor")` ?

Reply:
done